Họ tên: Lê Quang Vũ

MSSV: 1712209

# BT01: Giới thiệu CUDA

Bạn có thể tùy ý thêm/xóa cell. Đừng xóa mấy cell đề mục của mình là được.

## Câu 1

In [ ]:
!nvcc bt01_p1.cu -o bt01_p1

In [ ]:
!./bt01_p1 in.pnm out.pnm 16 16

Image size (width x height): 512 x 512

Processing time (use host): 2.301952 ms

GPU name: Tesla T4
GPU compute capability: 7.5
Processing time (use device): 0.770400 ms

Error between device result and host result: 0.002674


In [ ]:
!./bt01_p1 in.pnm out.pnm 32 32

Image size (width x height): 512 x 512

Processing time (use host): 2.352576 ms

GPU name: Tesla T4
GPU compute capability: 7.5
Processing time (use device): 0.752992 ms

Error between device result and host result: 0.002674


In [ ]:
!./bt01_p1 in.pnm out.pnm 64 64

Image size (width x height): 512 x 512

Processing time (use host): 2.264864 ms

GPU name: Tesla T4
GPU compute capability: 7.5
Processing time (use device): 0.756064 ms

Error between device result and host result: 119.969627


Độ lỗi khi chạy với các block 

- block = 16:  0.002674

- block = 32:  0.002674

- block = 64:  119.969627

## Câu 2

In [36]:
!nvcc bt01_p2.cu -o bt01_p2

In [37]:
!./bt01_p2 in.pnm out2.pnm out2_target.pnm 16 16

Image size (width x height): 512 x 512

Processing time (use host): 265.013245 ms
Error: 0.002693

GPU name: Tesla T4
GPU compute capability: 7.5
Processing time (use device): 1.255104 ms
Error: 0.002744



In [38]:
!./bt01_p2 in.pnm out2.pnm out2_target.pnm 32 32

Image size (width x height): 512 x 512

Processing time (use host): 260.475159 ms
Error: 0.002693

GPU name: Tesla T4
GPU compute capability: 7.5
Processing time (use device): 1.152192 ms
Error: 0.002744



In [39]:
!./bt01_p2 in.pnm out2.pnm out2_target.pnm 64 64

Image size (width x height): 512 x 512

Processing time (use host): 255.812134 ms
Error: 0.002693

GPU name: Tesla T4
GPU compute capability: 7.5
Processing time (use device): 0.885504 ms
Error: 124.033791

